static/app/views/settings/organizationIntegrations/integrationButton.tsx

```diff
@@ -0,0 +1,98 @@
+import {Fragment} from 'react';
+
+import {Button} from 'sentry/components/button';
+import {IconOpen} from 'sentry/icons';
+import type {
+  Integration,
+  IntegrationProvider,
+  IntegrationType,
+} from 'sentry/types/integrations';
+import type {Organization} from 'sentry/types/organization';
+import {AddIntegrationButton} from 'sentry/views/settings/organizationIntegrations/addIntegrationButton';
+import RequestIntegrationButton from 'sentry/views/settings/organizationIntegrations/integrationRequest/RequestIntegrationButton';
+
+type Props = {
+  analyticsParams: {
+    already_installed: boolean;
+    view:
+      | 'integrations_directory_integration_detail'
+      | 'integrations_directory'
+      | 'onboarding'
+      | 'project_creation';
+  };
+  buttonProps: ButtonProps;
+  installStatus: string;
+  onAddIntegration: (integration: Integration) => void;
+  onExternalClick: () => void;
+  organization: Organization;
+  provider: IntegrationProvider;
+  type: IntegrationType;
+  userHasAccess: boolean;
+  externalInstallText?: string;
+  modalParams?: {[key: string]: string};
+};
+
+type ButtonProps = {
+  disabled?;
+  priority?;
+  size?;
+  style?;
+} | null;
+
+function IntegrationButton({
+  onAddIntegration,
+  onExternalClick,
+  organization,
+  provider,
+  type,
+  userHasAccess,
+  installStatus,
+  analyticsParams,
+  externalInstallText,
+  modalParams = {},
+  buttonProps,
```

Comment:
I'd recommend adding a React.context to access the properties that aren't intended to be directly an attribute of a "button" - by that i mean things like data or adding to the parent state should ideally be accessed via a react.context. this allows you to share all of that data with all the children components of a single parent, and it means that the interface to this component will just look like a button (making it easier to reuse in the future)